raiden-ts/src/raiden.ts

```diff
@@ -705,37 +706,35 @@ export class Raiden {
    *
    * @param token - Token address on currently configured token network registry
    * @param target - Target address (must be getAvailability before)
-   * @param amount - Amount to try to transfer
+   * @param value - Amount to try to transfer
    * @param options - Optional parameters for transfer:
    *    <ul>
    *      <li>paymentId - payment identifier, a random one will be generated if missing</li>
    *      <li>secret - Secret to register, a random one will be generated if missing</li>
    *      <li>secrethash - Must match secret, if both provided, or else, secret must be
    *          informed to target by other means, and reveal can't be performed</li>
-   *      <li>metadata - Used to specify possible routes instead of querying PFS.</li>
+   *      <li>paths - Used to specify possible routes & fees instead of querying PFS.</li>
    *    </ul>
    * @returns A promise to transfer's secrethash (unique id) when it's accepted
    */
   public async transfer(
     token: string,
     target: string,
-    amount: BigNumberish,
+    value: BigNumberish,
     options: {
       paymentId?: BigNumberish;
       secret?: string;
       secrethash?: string;
-      metadata?: { readonly routes: { readonly route: string[] }[] };
+      paths?: RaidenPaths;
```

Comment:
As a developer I would prefer to pass `paths` as an argument consisting of only native, simple JavaScript types

Reply:
Yes, that's exactly what this `RaidenPaths` is about, it's a way to expose `Paths` type as simple JS types. It's not a codec, nor include our specialized types. Notice we use `BigNumberish` as just an union which includes all types which can be converted to `BigNumber`: `number | string | UInt8Array | BigNumber`. With the exception of the last, everything else is pure JS types, and accepting `BigNumber` directly is important if the developer has access to them (e.g. by importing from `ethers/utils`, as we do in the dApp).